bsp/raspberrypi/rp2040/src/hal/spi.zig

```diff
@@ -72,39 +156,104 @@ pub const SPI = enum(u1) {
     pub inline fn is_readable(spi: SPI) bool {
         return spi.get_regs().SSPSR.read().RNE == 1;
     }
-    pub fn transceive(spi: SPI, src: []const u8, dst: []u8) usize {
+
+    fn validate_bitwidth(PacketType: type) void {
+        const acceptable_types = .{ u4, u5, u6, u7, u8, u9, u10, u11, u12, u13, u14, u15, u16 };
+        for (acceptable_types) |acceptable_type| {
+            if (std.meta.eql(PacketType, acceptable_type)) return;
+        }
```

Comment:
```zig
if(@bitSizeOf(PacketType) < 4 or @bitSizeOf(PacketType) > 16)
  @compileError("PacketType must have between 4 and 16 bits");
```
This allows us to use arbitrary data items instead of forcing uints :)

Reply:
Ahhh nice ,this is what I was fumbling for when I cooked up this :)